src/app/models/project.model.ts

```diff
@@ -221,15 +223,63 @@ export class Project {
     return paid;
   }
 
-  getFairShares() {
+  setUnevenSplit(weights: {user: User, weight: number}[]) {
+    if(weights.length < this.users.length) {
+      console.warn('Not all users are receiving a weight. Users with unset weights will receive a value of 1');
+    }
+
+    this.weights = [];
+
+    this.users.forEach(user => {
+      let weight = weights.find(weight => weight.user.id === user.id);
+      if(weight) {
+        this.weights.push(weight);
+      } else {
+        this.weights.push({user, weight:1});
```

Comment:
A warn should be logged here to (same as in line 228), as it is also a situation where a user did not receive a weight.

Reply:
Also, for each user without a weight, it could save the name and then emit a warning with a message like "Users a, b, c did not receive weights..."